src/main/java/com/google/cloud/dataproc/templates/jdbc/JDBCToGCS.java

```diff
@@ -0,0 +1,171 @@
+            + "1. {}:{}"
+            + "2. {}:{}"
+            + "3. {}:{}"
+            + "4. {},{}"
+            + "5. {},{}"
+            + "6. {},{}",
+        JDBC_TO_GCS_OUTPUT_LOCATION,
+        GCSOutputLocation,
+        JDBC_TO_GCS_INPUT_TABLE_PROP,
+        jdbcInputTable,
+        JDBC_TO_GCS_INPUT_TABLE_DATABASE_PROP,
+        jdbcInputDb,
+        JDBC_TO_GCS_WRITE_MODE,
+        GCSWriteMode,
+        JDBC_TO_GCS_JDBC_URL,
+        jdbcURL);
+
+    try {
+      spark =
+          SparkSession.builder().appName("Spark JDBCToGCS Job").enableHiveSupport().getOrCreate();
+
+      Properties connectionProperties = new Properties();
+      List<String> jsonList = new ArrayList<>();
+      // Following code is to convert string json properties as values.
+      // TODO -- Replace deprecated function for reading json
+      jsonList.add(jdbcPropoertiesJSON);
+      JavaSparkContext javaSparkContext = new JavaSparkContext(spark.sparkContext());
+      JavaRDD<String> javaRdd = javaSparkContext.parallelize(jsonList);
+      Dataset<Row> jdbcPropertiesDF = spark.read().json(javaRdd);
+      String[] propertyNames = jdbcPropertiesDF.schema().fieldNames();
+      Row propertyValues = jdbcPropertiesDF.collectAsList().get(0);
+
+      // Adding user provided jdbc properties
+      int i = 0;
+      for (String propName : propertyNames) {
+        connectionProperties.setProperty(propName, propertyValues.getString(i));
+        i++;
+      }
+      connectionProperties.setProperty("driver", jdbcDriverClassName);
+
+      /** Read Input data from JDBC table */
+      Dataset<Row> inputData = null;
+
+      // If Input in specified as database and table
+      if ((!StringUtils.isAllBlank(jdbcInputTable) && !StringUtils.isAllBlank(jdbcInputDb))) {
+        inputData =
+            spark.read().jdbc(jdbcURL, jdbcInputDb + "." + jdbcInputTable, connectionProperties);
+      }
+
+      // If Input in specified as sql query
+      if (!StringUtils.isAllBlank(jdbcSQL)) {
+
+        DataFrameReader jdbcInputConfig =
+            spark.read().format("jdbc").option("url", jdbcURL).option("query", jdbcSQL);
+
+        // Adding all custom jdbc properties to config
+        for (String propName : connectionProperties.stringPropertyNames()) {
+          jdbcInputConfig =
+              jdbcInputConfig.option(propName, connectionProperties.getProperty(propName));
+        }
+
+        inputData = jdbcInputConfig.load();
+      }
+
+      DataFrameWriter<Row> writer = inputData.write().mode(GCSWriteMode).format(GCSOutputFormat);
+
+      /*
+       * If optional partition column is passed than partition data by partition
+       * column before writing to GCS.
+       * */
+      if (StringUtils.isNotBlank(GCSPartitionColumn)) {
+        LOGGER.info("Partitioning data by :{} cols", GCSPartitionColumn);
+        writer = writer.partitionBy(GCSPartitionColumn);
+      }
+
+      writer.save(GCSOutputLocation);
+
+    } catch (Throwable th) {
```

Comment:
Do we need this try / catch ?
I know we previously have done in several templates but I if we don't need it then lets remove it.

Reply:
Removed try catch and also verified that spark able to handle errors/exceptions and returning non-zero return code in failure.